### PR TITLE
fix(ceramic-node): handle case where the admin DID is not authenticated

### PR DESCRIPTION
**Pull request by gvelez17**
## Description

When running with simple deploy, we receive the error

```
Error: DID is not authenticated
    at get id [as id]
   ```
   
  handle this case and report the error, but do not fail to start

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [x ] Unit tests

## PR checklist

Before submitting this PR, please make sure:

- [ ] I have tagged the relevant reviewers and interested parties
- [ ] I have updated the READMEs of affected packages
- [ ] I have made corresponding changes to the documentation

## References:

Please list relevant documentation (e.g. tech specs, articles, related work etc.) relevant to this change, and note if the documentation has been updated.

### CODE REVIEW

File: packages/cli/src/ceramic-daemon.ts [hunk 1]
```diff
@@ -344,24 +344,34 @@ export class CeramicDaemon {
...
-      })
-      diagnosticsLogger.imp(
-        `Publishing Node Metrics publicly to the Ceramic Network.  To learn more, including how to disable publishing, please see the NODE_METRICS.md file for your branch, e.g. https://github.com/ceramicnetwork/js-ceramic/blob/develop/docs-dev/NODE_METRICS.md`
-      )
+      let didId: string | null = null;
+      try {
+        didId = did.id
+      } catch (err: any) {
+        diagnosticsLogger.imp(
+          `Unable to publish node metrics without an authenticated DID.  If you wish your ceramic node to be able to publish metrics a privateSeedUrl is required. Error: ${err.toString()}`
```
Comment (stbrody) [inline packages/cli/src/ceramic-daemon.ts:354]: `privateSeedUrl` is required, the Ceramic daemon won't start up without it.  You need the node DID to authenticate requests to the CAS.  So there's no way the node could not have a DID.  I think there must be something else going on here.

Comment (gvelez17) [reply packages/cli/src/ceramic-daemon.ts:354]: I don't know, but in fact it is not present, and if i add it, then it doesn't fail

`https://github.com/ceramicstudio/simpledeploy/blob/main/k8s/base/composedb/daemon-config.json#L25`

File: packages/cli/src/ceramic-daemon.ts [hunk 2]
```diff
@@ -344,24 +344,34 @@ export class CeramicDaemon {
...
-        nodeName: '', // daemon.hostname is not useful
-        nodeAuthDID: did.id,
-        nodeIPAddr: '', // daemon.hostname is not the external name
-        nodePeerId: ipfsId.publicKey,
-        logger: diagnosticsLogger,
-      })
-      diagnosticsLogger.imp(
-        `Publishing Node Metrics publicly to the Ceramic Network.  To learn more, including how to disable publishing, please see the NODE_METRICS.md file for your branch, e.g. https://github.com/ceramicnetwork/js-ceramic/blob/develop/docs-dev/NODE_METRICS.md`
-      )
+      let didId: string | null = null;
```
Comment (smrz2001) [inline packages/cli/src/ceramic-daemon.ts:349]: Seems like we can simplify this logic. Can we just check if `did.id` is set and then use that in the `if` check below?

Comment (gvelez17) [reply packages/cli/src/ceramic-daemon.ts:349]: no, because the getter for did.id is what throws an error

Comment (gvelez17) [reply packages/cli/src/ceramic-daemon.ts:349]: @smrz2001 this is where the error was being thrown, I believe we have to catch it: https://github.com/ceramicnetwork/js-did/blob/main/packages/dids/src/did.ts#L192

Comment (smrz2001) [reply packages/cli/src/ceramic-daemon.ts:349]: Can we use the `authenticated` method instead?

File: packages/cli/src/ceramic-daemon.ts [hunk 3]
```diff
@@ -344,24 +344,26 @@ export class CeramicDaemon {
...
-        nodeName: '', // daemon.hostname is not useful
-        nodeAuthDID: did.id,
-        nodeIPAddr: '', // daemon.hostname is not the external name
-        nodePeerId: ipfsId.publicKey,
-        logger: diagnosticsLogger,
-      })
-      diagnosticsLogger.imp(
-        `Publishing Node Metrics publicly to the Ceramic Network.  To learn more, including how to disable publishing, please see the NODE_METRICS.md file for your branch, e.g. https://github.com/ceramicnetwork/js-ceramic/blob/develop/docs-dev/NODE_METRICS.md`
-      )
+      if (did.authenticated) {
```
Comment (stbrody) [inline packages/cli/src/ceramic-daemon.ts:349]: nice cleanup, I do still think logging a warning when there isn't a node DID is a good idea though.

**Review by stbrody — APPROVED** [review]
lgtm

**Review by smrz2001 — APPROVED** [review]
🥳

**Comment by gvelez17** [comment]
huh i thought i saw where @stbrody you wanted a warning about not being authenticated, so i changed it to add the warning - may merge on prior approvals tho since i want to get this in

**Pull request merged**